include/lsst/afw/detection/Psf.h

```diff
@@ -283,13 +298,29 @@ class Psf : public daf::base::Citizen,
     virtual geom::Box2I doComputeBBox(geom::Point2D const& position, image::Color const& color) const = 0;
     //@}
 
+    struct CacheKey {
+        geom::Point2D position;
+        image::Color color;
+
+        CacheKey(geom::Point2D const& position_, image::Color color_=image::Color())
+          : position(position_), color(color_) {}
+
+        friend std::size_t hash_value(CacheKey const& key) {
+            return boost::hash<geom::Point2D>()(key.position);  // Currently don't care about color
+        }
+
+        bool operator==(CacheKey const& other) const {
+            return position == other.position;  // Currently don't care about color
+        }
+
+        friend std::ostream & operator<<(std::ostream &os, CacheKey const &key) {
+            return os << key.position;
+        }
+    };
+
```

Comment:
Hopefully this can be moved to the .cc file, too.

Reply:
Pulled it out of the class, into a preceding `namespace detail`. But it does mess up the top of the file a bit, especially since I have to get into `namespace std` to provide the `std::hash` support (it's more annoying to work with than `boost::hash`).